controllers/cloud.redhat.com/providers/metrics/operator.go

```diff
@@ -0,0 +1,202 @@
+package metrics
+
+import (
+	"fmt"
+
+	crd "cloud.redhat.com/clowder/v2/apis/cloud.redhat.com/v1alpha1"
+	"cloud.redhat.com/clowder/v2/controllers/cloud.redhat.com/config"
+	"cloud.redhat.com/clowder/v2/controllers/cloud.redhat.com/providers"
+	"cloud.redhat.com/clowder/v2/controllers/cloud.redhat.com/utils"
+	prom "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	core "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type metricsProvider struct {
+	providers.Provider
+}
+
+var PrometheusInstance = providers.NewSingleResourceIdent(ProvName, "prometheus_instance", &prom.Prometheus{})
+
+var PrometheusRole = providers.NewSingleResourceIdent(ProvName, "prometheus_role", &rbac.Role{})
+
+var PrometheusRoleBinding = providers.NewSingleResourceIdent(ProvName, "prometheus_role_binding", &rbac.RoleBinding{})
+
+var PrometheusServiceAccount = providers.NewSingleResourceIdent(ProvName, "prometheus_service_account", &core.ServiceAccount{})
+
+func NewMetricsProvider(p *providers.Provider) (providers.ClowderProvider, error) {
+	if !p.Env.Spec.Providers.Metrics.Prometheus.Deploy {
+		return &metricsProvider{Provider: *p}, nil
+	}
+
+	promObj := &prom.Prometheus{}
+	nn := types.NamespacedName{
+		Name:      p.Env.Name,
+		Namespace: p.Env.Status.TargetNamespace,
+	}
+	if err := p.Cache.Create(PrometheusInstance, nn, promObj); err != nil {
+		return nil, err
+	}
+
+	promObj.SetName(p.Env.Name)
```

Comment:
It doesn't really matter, but do we want to use `nn.Name` and `Namespace` here?

Reply:
We could do for sure - :)